pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -228,8 +228,9 @@ func (rc *reconciler) reconcile() {
 
 			// Trigger detach volume which requires verifying safe to detach step
 			// If timeout is true, skip verifySafeToDetach check
+			// If the node has node.kubernetes.io/out-of-service taint with NoExecute effect, skip verifySafeToDetach check
 			klog.V(5).InfoS("Starting attacherDetacher.DetachVolume", "volume", attachedVolume)
-			verifySafeToDetach := !timeout
+			verifySafeToDetach := !timeout && !hasOutOfServiceTaint
```

Comment:
This change is necessary to make sure that when the pod gets on to a healthy node it starts running immediately and does not stay stuck in `ContainerCreating` state for 6 min.

Reply:
From a readability POV, I would suggest something like:

```
knownSafeToDetach := timeout || hasOutOfServiceTaint
err = rc.attacherDetacher.DetachVolume(attachedVolume.AttachedVolume, !knownSafeToDetach, rc.actualStateOfWorld)
```

or 

```
verifySafeToDetach := !(timeout || hasOutOfServiceTaint)
```